lib/Pipes/PipelineManager.cpp

```diff
@@ -448,3 +448,35 @@ PipelineManager::invalidateFromDiff(const llvm::StringRef Name,
   // TODO: once invalidations are working, return `Map` instead of this
   return invalidateAllPossibleTargets();
 }
+
+llvm::Error PipelineManager::produceTargets(const llvm::StringRef StepName,
+                                            const Container &TheContainer,
+                                            const TargetsList &List,
+                                            llvm::raw_ostream &Output) {
+  if (CurrentState.count(StepName) == 0)
```

Comment:
CurrentState.count(StepName) == 0 implies that no target can be produced is the given Step, not that no such step exists. 

while the code in the commit kind of works, the error message will some time be confusing. Current state is a table that contains the list of produced targets, it does not wishes to properly keep track of whch steps exists (even tough maybe the current implementation may happen to guarantee it). You must interrogate the runner to check if StepName exists, not CurrentState.